### PR TITLE
Dependency refactor

### PR DESCRIPTION
**Pull request by melcz**
Removing guava dependencies from dexlib2/analysis

### CODE REVIEW

File: dexlib2/src/main/java/com/android/tools/smali/dexlib2/analysis/ArrayProto.java [hunk 1]
```diff
@@ -141,7 +140,7 @@ public TypeProto getCommonSuperclass(@Nonnull TypeProto other) {
         return other.getCommonSuperclass(this);
     }
 
-    private static final String BRACKETS = Strings.repeat("[", 256);
+    private static final String BRACKETS = "[".repeat(256);
```
Comment (sgjesse) [inline dexlib2/src/main/java/com/android/tools/smali/dexlib2/analysis/ArrayProto.java:143]: [`String.repeat`](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/lang/String.html#repeat(int)) was not added until JDK-11.

Can we add a local `StringUtil` implementation of this? Maybe just copy the Guava optimized implementation, https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Strings.java#L152, or as this is the only use just a simple loop here.

File: dexlib2/src/main/java/com/android/tools/smali/dexlib2/analysis/util/MemoizingSupplier.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        if (delegate == null) {
+            throw new NullPointerException("delegate == null");
+        }
+        this.delegate = delegate;
+    }
+
+    @Override
+    public T get() {
+        // Because Supplier is read-heavy, we use the "double-checked locking" pattern.
+        if (delegate != SUCCESSFULLY_COMPUTED) {
```
[the remark this answers is not in the document]
Comment (melcz) [reply dexlib2/src/main/java/com/android/tools/smali/dexlib2/analysis/util/MemoizingSupplier.java]: changed, and added a comment

File: dexlib2/src/main/java/com/android/tools/smali/dexlib2/analysis/ArrayProto.java (hunk 1 above)
Comment (melcz) [reply dexlib2/src/main/java/com/android/tools/smali/dexlib2/analysis/ArrayProto.java]: Ported into StringUtils

**Review by sgjesse — APPROVED** [review]
Thank you!

**Pull request merged**